### DISCUSSION TITLE
How to Check mmengine version?

### DISCUSSION THREAD

**Discussion by MatCorr** [open-mmlab/mmengine#1580] [Q&A] [answered]
I have mmsegmentation installed and running successfully, which means (I think) that mmengine is also installed.

However, recently, I came across the need to check my mmengine version and it has been proving to be harder than expected.

I've tried running the following command `python -c 'import mmengine;print(mmengine.__version__)'` and received this as a result:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'mmengine'
```
And I've also tried this `mim list` with no luck:

```
Package         Version    Source
--------------  ---------  ----------------------------------------------
mmcls           0.25.0     https://github.com/open-mmlab/mmclassification
mmcv-full       1.5.0      https://github.com/open-mmlab/mmcv
mmdet           2.28.1     https://github.com/open-mmlab/mmdetection
mmsegmentation  0.27.0     http://github.com/open-mmlab/mmsegmentation
```

So, what can I do? Thanks for the help!

**Comment by fanqiNO1** [comment, marked as answer]
This problem is caused by the fact that mmdet 2.x and mmseg 0.x **do not** depend on mmengine, so a successful run of mmseg does not indicate that mmengine is installed. In fact, you don't have mmengine installed, so perhaps you can install it via `pip install mmengine`.

**Reply by MatCorr** [reply]
Thanks a lot!

**Comment by Alhanoufalsenan** [comment]
mmengine worked with my code but now it shows me this message ''ASGD is already registered in optimizer at torch.optim.asgd''.
so, please can someone advise me how to fix this problem
